docarray/array/storage/redis/seqlike.py

```diff
@@ -0,0 +1,80 @@
+from typing import Iterable, Union
+
+from .... import Document
+from ..base.seqlike import BaseSequenceLikeMixin
+
+
+class SequenceLikeMixin(BaseSequenceLikeMixin):
+    """Implement sequence-like methods for DocumentArray with Redis as storage"""
+
+    def __eq__(self, other):
+        """Compare this object to the other, returns True if and only if other
+        as the same type as self and other has the same meta information
+
+        :param other: the other object to check for equality
+        :return: ``True`` if other is equal to self
+        """
+        # two DA are considered as the same if they have the same client meta data
+        return (
+            type(self) is type(other)
+            and self._client.client_info() == other._client.client_info()
+            and self._config == other._config
+        )
+
+    def __len__(self):
+        """Return the length of :class:`DocumentArray` that uses Redis as storage
+
+        :return: the length of this :class:`DocumentArrayRedis` object
+        """
+        try:
+            keys = self._client.keys()
+            if b'offset2id' in keys:
+                return len(keys) - 1
+            else:
+                return len(keys)
```

Comment:
This will return all keys in the database (image we have millions of those)
let's instead use `dbsize()` and check `b'offset2id' in keys` differently (for example client.exists or so)